etl/etl.py

```diff
@@ -3,51 +3,105 @@
 import pathlib
 from pprint import pprint
 from contextlib import closing
+from typing import List, Optional
 
 from gremlin_python.process.anonymous_traversal import traversal
 from gremlin_python.driver.driver_remote_connection import DriverRemoteConnection
 from gremlin_python.process.graph_traversal import __
 
-def extract_from_nix():    
+
+def extract_from_nix():
     nixpkgs_graph_nix_file_path = (
         pathlib.Path(__file__).parent.joinpath("nixpkgs-graph.nix").resolve()
     )
-    print("Downloading data from nix as a dataframe...")    
-    result = subprocess.run('NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file '+ str(nixpkgs_graph_nix_file_path) + ' --extra-experimental-features "nix-command flakes"', shell=True, stdout=subprocess.PIPE)
-    dataframe = pd.read_json(result.stdout.decode(), orient="records")    
+    print("Downloading data from nix as a dataframe...")
+    result = subprocess.run(
```

Comment:
I think we might want to add a `check=True` to the `subprocess.run()` call to ensure that this fails fast (rather than failing with a cryptic error when attempting to decode stdout)